tests/core/actions/test_forms.py

```diff
@@ -1334,3 +1358,73 @@ async def test_ask_for_slot_if_not_utter_ask(
 
     assert not events
     action_from_name.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "global_not_intent, slot_not_intent",
+    [
+        (
+            # `global_not_intent` as a string and slot's not_intent as an empty list.
+            "greet",
+            [],
+        ),
+        (
+            # `global_not_intent` as an empty list and slot's not_intent has a value.
+            [],
+            ["greet"],
+        ),
+        (
+            # `global_not_intent` as a list of 2 values and slot's not_intent has one
+            # value different than the ones in `global_not_intent`.
+            ["chitchat", "greet"],
+            ["inform"],
+        ),
+        (
+            # `global_not_intent` as a list of 2 values and slot's not_intent has one
+            # value that is included also in `global_not_intent`.
+            ["chitchat", "greet"],
+            ["chitchat"],
+        ),
+    ],
+)
+async def test_global_not_intent(
```

Comment:
I think we don't need `async` here, do we? 
```suggestion
def test_global_not_intent(
```